packages/experimental/src/experimental-list/collapsible-list/index.tsx

```diff
@@ -48,15 +90,75 @@ export const ExperimentalCollapsibleList: React.FC< CollapsibleListProps > = ( {
 } ): JSX.Element => {
 	const [ isCollapsed, setCollapsed ] = useState( collapsed );
 	const [ containerHeight, setContainerHeight ] = useState( 0 );
+	const [ footerLabels, setFooterLabels ] = useState( {
+		collapse: collapseLabel,
+		expand: expandLabel,
+	} );
+	const [ displayedChildren, setDisplayedChildren ] = useState< {
+		all: React.ReactElement[];
+		shown: React.ReactElement[];
+		hidden: React.ReactElement[];
+	} >( {
+		all: [],
+		shown: [],
+		hidden: [],
+	} );
 	const collapseContainerRef = useCallback(
 		( containerElement: HTMLDivElement ) => {
 			if ( containerElement ) {
 				setContainerHeight( getContainerHeight( containerElement ) );
 			}
 		},
-		[ children ]
+		[ displayedChildren.hidden ]
 	);
 
+	const updateChildren = () => {
+		let shownChildren: React.ReactElement[] = [];
+		const allChildren =
+			Children.map( children, ( child ) =>
+				isValidElement( child ) && 'key' in child ? child : null
+			) || [];
+		let hiddenChildren = allChildren;
+		if ( show > 0 ) {
+			shownChildren = allChildren.slice( 0, show );
+			hiddenChildren = allChildren.slice( show );
+		}
+		if ( hiddenChildren.length > 0 ) {
+			setFooterLabels( { expand: expandLabel, collapse: collapseLabel } );
```

Comment:
Is this required? It seems like it won't change from the initial values.

Reply:
Not overly, I did it because of the staggered transition. Because during the transition you would see the label change to `Show 0 items`, which looked a little odd. This way it keeps the same label when it minimizes.
I could push that logic upwards otherwise.

Reply:
A comment in the code would suffice here - thanks for the explanation.